addons/html_editor/static/src/core/base_container_plugin.js

```diff
@@ -167,4 +174,37 @@ export class BaseContainerPlugin extends Plugin {
             }
         }
     }
+
+    /**
+     * Transform an empty heading, blockquote, pre or signature container
+     * at the beginning of the editable into a base container.
+     */
+    handleDeleteBackward({ startContainer, startOffset, endContainer, endOffset }) {
+        // Detect if cursor is at the start of the editable (collapsed range).
+        const rangeIsCollapsed = startContainer === endContainer && startOffset === endOffset;
+        if (!rangeIsCollapsed) {
+            return;
+        }
+        const block = closestBlock(endContainer);
+        // Check if the block is not empty.
+        if (!isEmptyBlock(block)) {
+            return;
+        }
+
+        // Check if the block is not transformable to a base container
+        // or is unremovable.
+        if (
+            !this.getResource("block_to_base_container_predicates").some((p) => p(block)) ||
+            this.getResource("unremovable_node_predicates").some((p) => p(block))
```

Comment:
you should probably use `isCandidateForBaseContainer` to replace `this.getResource("unremovable_node_predicates").some((p) => p(block))` unless there are known specific cases where it is too restrictive, in that case it would be interesting to investigate these exceptions.